pandas/util/testing.py

```diff
@@ -1049,6 +1051,18 @@ def assert_period_array_equal(left, right, obj='PeriodArray'):
     assert_attr_equal('freq', left, right, obj=obj)
 
 
+def assert_datetime_array_equal(left, right, obj='DatetimeArray'):
+    _check_isinstance(left, right, DatetimeArray)
+
+    assert_numpy_array_equal(left._data, right._data,
+                             obj='{obj}._data'.format(obj=obj))
+    assert_attr_equal('freq', left, right, obj=obj)
+    assert_attr_equal('tz', left, right, obj=obj)
+
+    # Check that == works as expected
+    assert ((left == right) | (left._isnan & right._isnan)).all()
```

Comment:
why is this needed? (we shouldn't be testing the equality operator inside the assert? or otherwise simply use this as the initial assert if it is sufficient?)

Reply:
I included this as a double-check because ATM the comparison operations on DatetimeArray are broken.  When we have a robust set of tests for those comparisons, this will be entirely unnecesary.  I don't mind removing it if it bothers you, but would personally prefer to leave it until those tests are in place.

Reply:
What do you mean with "comparison operations are broken" ?

Reply:
> What do you mean with "comparison operations are broken" ?

on master:

```
dti = pd.date_range('2016-01-01', periods=3)
arr = DatetimeArrayMixin(dti)

>>> arr == arr
array([False, False, False])

>>> arr < dti
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/arrays/datetimes.py", line 125, in wrapper
    return ops.invalid_comparison(self, other, op)
  File "pandas/core/ops.py", line 913, in invalid_comparison
    .format(dtype=left.dtype, typ=type(right).__name__))
TypeError: Invalid comparison between dtype=datetime64[ns] and DatetimeArrayMixin
```

This, others in the arithmetic ops, and I'm assuming others I haven't found yet, are why I am so adamant that we get the tests working on DTA/TDA as stand-alone objects before worrying about the composition and/or constructor names.

Reply:
I see.
Still, I would simply remove the check here already now. We will shortly make sure the comparison operations work, and will add proper tests then.

(still I would

Reply:
done